### PR TITLE
 addet pagination,Sorting, Swagger

### PR DESCRIPTION
**Pull request by DmytroHadiuchko**

### CODE REVIEW

File: src/main/java/dmytro/hadiuchko/springboot/controller/BookController.java [hunk 1]
```diff
@@ -18,41 +21,50 @@
...
     public void updateById(@PathVariable Long id,
                            @Valid @RequestBody CreateBookRequestDto bookRequestDto) {
         bookService.updateById(id, bookRequestDto);
     }
 
     @GetMapping("/search")
-    public List<BookDto> search(@Valid BookSearchParametersDto searchParameters) {
-        return bookService.search(searchParameters);
+    @Operation(summary = "search book according param", description = "filter and returns "
+            + "books according param")
```
Comment (andrii-hoienko) [inline src/main/java/dmytro/hadiuchko/springboot/controller/BookController.java]:
```suggestion
    @Operation(summary = "search book according to search parameters", description = "filter and returns "
            + "books according to search parameters")

```

File: pom.xml [hunk 2]
```diff
@@ -46,6 +46,11 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-validation</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
+            <version>2.1.0</version>
```
Comment (andrii-hoienko) [inline pom.xml]: Please use variable to define version like you did n other dependencies

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, left minor comments

File: src/main/java/dmytro/hadiuchko/springboot/controller/BookController.java [hunk 3]
```diff
@@ -18,41 +21,50 @@
...
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all products", description = "Get list with sorted products")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "get product by id", description = "return book by id")
```
Comment (yevheniisynytsia) [inline src/main/java/dmytro/hadiuchko/springboot/controller/BookController.java]:
```suggestion
    @Operation(summary = "Get product by id", description = "Return book by id")
```
let's make those pretty

File: src/main/java/dmytro/hadiuchko/springboot/controller/BookController.java [hunk 4]
```diff
@@ -18,41 +21,50 @@
...
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.findById(id);
     }
 
     @PostMapping
     @ResponseStatus(value = HttpStatus.CREATED)
+    @Operation(summary = "create a new book", description = "create a new book")
```
Comment (yevheniisynytsia) [inline src/main/java/dmytro/hadiuchko/springboot/controller/BookController.java]: same for others

**Review by yevheniisynytsia — APPROVED** [review]

**Pull request merged**